xds/server_options.go

```diff
@@ -0,0 +1,85 @@
+/*
+ *
+ * Copyright 2021 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package xds
+
+import (
+	"net"
+
+	"google.golang.org/grpc"
+)
+
+// Experimental
+//
+// Notice: All APIs in this file are EXPERIMENTAL and may be changed or removed
+// in a later release.
```

Comment:
This won't appear in godoc, so is meaningless.  All individual types/functions should be marked with this, since we can't put it at the package level.

I wonder if it makes sense to document this at the package level anyway?  The package itself isn't experimental, but all APIs within it are.

Reply:
> Does it matter that the resolvers and balancers are automatically available when users blank import the package, and is that why we might want to put it at the package level?

This is the only reason I think we _wouldn't_ put it at the package level.  Because the package itself and its side-effects from importing are stable.  But all the APIs it contains are not.  I think we could add it there, and just make sure it's explicit that "the APIs in this package are experimental".